pallets/fusion/src/lib.rs

```diff
@@ -0,0 +1,3044 @@
+	#[pallet::getter(fn fusion_currencies)]
+	pub type FusionCurrencies<T: Config> =
+		StorageMap<_, Twox64Concat, CurrencyId, FusionCurrency<T>, OptionQuery>;
+
+	/// Stores all the fusion pools
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_pools)]
+	pub type FusionPools<T: Config> =
+		StorageMap<_, Twox64Concat, PoolId, FusionPool<T>, OptionQuery>;
+
+	/// Mapping from the pools funds account address to the pool id
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_pool_account_to_id)]
+	pub type FusionPoolsAccountToId<T: Config> =
+		StorageMap<_, Twox64Concat, T::AccountId, PoolId, OptionQuery>;
+
+	/// Stores all the membership of users in pools
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_memberships)]
+	pub type FusionMemberships<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		EvmAddress,
+		Twox64Concat,
+		PoolId,
+		FusionMembership<T>,
+		OptionQuery,
+	>;
+
+	/// Stores all the users idle balances
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_member_currency_balances)]
+	pub type FusionUserCurrencyBalances<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		EvmAddress,
+		Twox64Concat,
+		CurrencyId,
+		FusionUserCurrencyBalance,
+		OptionQuery,
+	>;
+
+	/// Stores era rewards for each pool
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_era_rewards)]
+	pub type FusionEraRewards<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		EraIndex,
+		Twox64Concat,
+		PoolId,
+		EraReward<T>,
+		OptionQuery,
+	>;
+
+	/// Stores the conversion rates for currencies
+	/// How much one unit of currency is equal in AVAIL
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_currency_rates)]
+	pub type FusionCurrencyRates<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		EraIndex,
+		Twox64Concat,
+		CurrencyId,
+		BalanceOf<T>,
+		OptionQuery,
+	>;
+
+	/// Stores the next currency changes to be applied next era
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_currency_rate_changes)]
+	pub type FusionCurrencyRateChanges<T: Config> =
+		StorageMap<_, Twox64Concat, CurrencyId, BalanceOf<T>, OptionQuery>;
+
+	/// Mapping from EVM Address to Substrate address
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_evm_to_substrate_address)]
+	pub type FusionEVMToSubstrateAddress<T: Config> =
+		StorageMap<_, Blake2_128Concat, EvmAddress, T::AccountId, OptionQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn era_durations)]
+	pub type EraDurations<T: Config> = StorageMap<_, Twox64Concat, EraIndex, u64, OptionQuery>;
+
+	/// Stores the fusion era exposure for HistoryDepth eras
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_era_data)]
+	pub type FusionExposures<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		EraIndex,
+		Twox64Concat,
+		PoolId,
+		FusionExposure<T>,
+		OptionQuery,
+	>;
+
+	/// Store the pools that backed the validator set as the key
+	#[pallet::storage]
+	#[pallet::getter(fn fusion_pools_from_validator)]
+	pub type FusionPoolsFromValidator<T: Config> = StorageDoubleMap<
```

Comment:
can we rename this to more meaningful like `ValidatorBackingPools` or something else, little confusing to me right now? Also, `set` in the doc is confusing

Reply:
Done but used the opposite for clarity: PoolsBackingValidator